### PR TITLE
Broadcast AbstractRNG like a scalar

### LINKED ISSUE #31071: AbstractRNG should be a broadcast scalar
As mentioned [on the mailing list](https://discourse.julialang.org/t/is-there-a-reason-a-mersennetwister-value-is-not-broadcastable/20774), we need a
```jl
Base.broadcastable(x::AbstractRNG) = Ref(x)
```
somewhere.

Not breaking since trying to a broadcast an RNG currently gives an error.

### PR DESCRIPTION
**Pull request by nickrobinson251**
- closes #31071 
- i think this is in-keeping with the de-facto heuristic on how things broadcast, which i think of as "something that's obviously not a collection broadcasts like a scalar, as long as there's a reasonable use-case for that behaviour"
- ideally we'd test this on a new/test-only AbstractRNG, but i wasn't sure exactly what would need to be implemented and if it'd be worth it... so this is just tested on the two subtypes definited in the module :)

### CODE REVIEW

File: NEWS.md [hunk 1]
```diff
@@ -41,11 +41,13 @@ Standard library changes
 
 * `dot` now admits a 3-argument method `dot(x, A, y)` to compute generalized dot products `dot(x, A*y)`, but without computing and storing the intermediate result `A*y` ([#32739]).
 
-#### SparseArrays
+#### Random
 
+* `AbstractRNG`s now behave like scalars when used in broadcasting ([#33213]).
```
Comment (oxinabox) [inline NEWS.md]:
```suggestion
* `AbstractRNG`s now behave like scalars when used in broadcasting ([#31071]).
```
I think convention is for the link to be to the issue that is closed,
and only to the PR if there is no issue.

I am not sure though, so wait for somone else to chime in

Comment (nickrobinson251) [reply NEWS.md]: I do not know - and would like to be advised

But if it should link to the issue (not the PR) we should update https://github.com/JuliaLang/julia/blob/master/CONTRIBUTING.md#news-worthy-changes

Comment (fredrikekre) [reply NEWS.md]: Why not both?

Comment (nickrobinson251) [reply NEWS.md]:
```suggestion
* `AbstractRNG`s now behave like scalars when used in broadcasting ([#31071, #33213]).
```

Comment (stevengj) [reply NEWS.md]: We normally link just the PR, I think.

File: NEWS.md [hunk 2]
```diff
@@ -41,11 +41,13 @@ Standard library changes
 
 * `dot` now admits a 3-argument method `dot(x, A, y)` to compute generalized dot products `dot(x, A*y)`, but without computing and storing the intermediate result `A*y` ([#32739]).
 
-#### SparseArrays
+#### Random
 
+* `AbstractRNG`s now behave like scalars when used in broadcasting ([#31071, #33213]).
```
Comment (stevengj) [inline NEWS.md]: This won't work.  The issue links have to be in the `[#nnnnnn]` format links to be generated.

I think that we normally just link to the PR in NEWS, so `[#33213]` should be sufficient here

**Review by stevengj — CHANGES_REQUESTED** [review]

File: NEWS.md (hunk 1 above)
Comment (nickrobinson251) [reply NEWS.md]: changed back

Comment (oxinabox) [reply NEWS.md]: My bad. Good to know

**Review by stevengj — APPROVED** [review]

**Review by stevengj — APPROVED** [review]

**Pull request merged**